packages/parser/src/ethereum/thor.ts

```diff
@@ -0,0 +1,103 @@
+import { ethers } from 'ethers'
+import { Tx } from '@shapeshiftoss/blockbook'
+import { Thorchain } from '@shapeshiftoss/thorchain'
+import { Dex, TxSpecific as ParseTxSpecific, TradeType } from '../types'
+import { Network } from './types'
+import ABI from './abi/thor'
+import { getSigHash } from './utils'
+
+const SWAP_TYPES = ['SWAP', '=', 's']
+
+export interface ParserArgs {
+  midgardUrl: string
+  network: Network
+}
+
+export class Parser {
+  abiInterface: ethers.utils.Interface
+  thorchain: Thorchain
+
+  readonly depositSigHash: string
+  readonly transferOutSigHash: string
+  readonly routerContract: string
+
+  constructor(args: ParserArgs) {
+    this.abiInterface = new ethers.utils.Interface(ABI)
+    this.thorchain = new Thorchain({ midgardUrl: args.midgardUrl })
+
+    this.depositSigHash = this.abiInterface.getSighash('deposit')
+    this.transferOutSigHash = this.abiInterface.getSighash('transferOut')
+
+    // TODO: Router contract can change, use /inbound_addresses endpoint to determine current router contract
+    this.routerContract = {
```

Comment:
how is this going to work over time? won't we need to maintain a mapping of inbound_addresses over time?

Reply:
For this method, yes you are correct. And I am not sure if there is actually a way to get past router contracts if you weren't aware of it at the time.